xarray/backends/netCDF4_.py

```diff
@@ -359,10 +355,7 @@ def open(
 
         if lock is None:
             if mode == "r":
-                if is_remote_uri(filename):
-                    lock = NETCDFC_LOCK
-                else:
-                    lock = NETCDF4_PYTHON_LOCK
+                lock = NETCDFC_LOCK if is_remote_uri(filename) else NETCDF4_PYTHON_LOCK
```

Comment:
This isn't really any better from my perspective. I guess it's one less token but mostly it's just the previous stuff re-arranged.

Reply:
Also, since this is now a single line, it is harder to tell if both branches are tested.